skglm/utils/prox_funcs.py

```diff
@@ -60,15 +60,26 @@ def value_MCP(w, alpha, gamma):
 
 
 @njit
-def prox_MCP(value, stepsize, alpha, gamma):
-    """Compute the proximal operator of stepsize * MCP penalty."""
+def value_weighted_MCP(w, alpha, gamma, weights):
+    """Compute the value of the weighted MCP."""
+    s0 = np.abs(w) < gamma * alpha
+    value = np.zeros_like(w)
+    value[~s0] = weights[~s0] * gamma * alpha ** 2 / 2.
+    value[s0] = alpha * np.abs(w[s0]) - w[s0]**2 / (2 * gamma)
+    value[s0] *= weights[s0]
+    return np.sum(value)
```

Comment:
Since the weighted MCP has the following formula

$$
\sum_j w_j \ \mathrm{MCP}(\beta_j)
$$

Its implementation should resemble `value_MCP` with the exception of having a weighted sum in the end.

```suggestion
def value_weighted_MCP(w, alpha, gamma, weights):
    """Compute the value of the weighted MCP."""
    s0 = np.abs(w) < gamma * alpha
    value = np.full_like(w, gamma * alpha ** 2 / 2.)
    value[s0] = alpha * np.abs(w[s0]) - w[s0]**2 / (2 * gamma)
    return np.sum(weights * value)
```